src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -32,73 +31,399 @@ const ActorInit En_Bubble_InitVars = {
+        return -1;
+    } else {
+        return func_8089F5D0(this);
+    }
+}
+
+s32 func_8089F908(EnBubble* this) {
+    this->modelWidth += 1.0f / 12.0f;
+    this->modelHeight += 1.0f / 12.0f;
+
+    if (DECR(this->explosionCountdown) != 0) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+void EnBubble_Vec3fNormalizedReflect(Vec3f* vec1, Vec3f* vec2, Vec3f* dest) {
+    f32 norm;
+
+    func_80179F64(vec1, vec2, dest);
+    norm = sqrtf(SQ(dest->x) + SQ(dest->y) + SQ(dest->z));
+    if (norm != 0.0f) {
+        dest->x /= norm;
+        dest->y /= norm;
+        dest->z /= norm;
+    } else {
+        dest->x = dest->y = dest->z = 0.0f;
+    }
+}
+
+void EnBubble_Vec3fNormalize(Vec3f* vec) {
+    f32 norm = sqrtf(SQ(vec->x) + SQ(vec->y) + SQ(vec->z));
+
+    if (norm != 0.0f) {
+        vec->x /= norm;
+        vec->y /= norm;
+        vec->z /= norm;
+    } else {
+        vec->x = vec->y = vec->z = 0.0f;
+    }
+}
+
+void EnBubble_Fly(EnBubble* this, PlayState* play) {
+    CollisionPoly* poly;
+    Actor* bumpActor;
+    Vec3f sp84;
+    Vec3f sp78;
+    Vec3f sp6C;
+    Vec3f normal;
+    Vec3f bounceDirection;
+    f32 bounceSpeed;
+    s32 bgId;
+    u8 bounceCount;
+
+    if (this->colliderSphere.elements[1].info.bumperFlags & BUMP_HIT) {
+        bumpActor = this->colliderSphere.base.ac;
+        this->normalizedBumpVelocity = bumpActor->velocity;
+        EnBubble_Vec3fNormalize(&this->normalizedBumpVelocity);
+        this->velocityFromBump.x += (this->normalizedBumpVelocity.x * 3.0f);
+        this->velocityFromBump.y += (this->normalizedBumpVelocity.y * 3.0f);
+        this->velocityFromBump.z += (this->normalizedBumpVelocity.z * 3.0f);
+    }
+    this->yVelocity -= 0.1f;
+    if (this->yVelocity < this->actor.terminalVelocity) {
+        this->yVelocity = this->actor.terminalVelocity;
+    }
+    bounceDirection.x = this->velocityFromBounce.x + this->velocityFromBump.x;
+    bounceDirection.y = this->velocityFromBounce.y + this->velocityFromBump.y + this->yVelocity;
+    bounceDirection.z = this->velocityFromBounce.z + this->velocityFromBump.z;
+    EnBubble_Vec3fNormalize(&bounceDirection);
+
+    sp78.x = this->actor.world.pos.x;
+    sp78.y = this->actor.world.pos.y + this->actor.shape.yOffset;
+    sp78.z = this->actor.world.pos.z;
+    sp6C = sp78;
+
+    sp6C.x += (bounceDirection.x * 24.0f);
+    sp6C.y += (bounceDirection.y * 24.0f);
+    sp6C.z += (bounceDirection.z * 24.0f);
+    if (BgCheck_EntityLineTest1(&play->colCtx, &sp78, &sp6C, &sp84, &poly, true, true, true, false, &bgId)) {
+        normal.x = COLPOLY_GET_NORMAL(poly->normal.x);
+        normal.y = COLPOLY_GET_NORMAL(poly->normal.y);
+        normal.z = COLPOLY_GET_NORMAL(poly->normal.z);
+        EnBubble_Vec3fNormalizedReflect(&bounceDirection, &normal, &bounceDirection);
+        this->bounceDirection = bounceDirection;
+        bounceCount = this->bounceCount;
+        this->bounceCount = ++bounceCount;
```

Comment:
Could you post a decompme scratch? I'm wondering if the increment is part of the if conditional.

Reply:
As mentioned in Discord, can't seem to find a way to make the increment part of the conditional.